qiskit/tools/visualization/_counts_visualization.py

```diff
@@ -17,35 +17,68 @@
 import matplotlib.pyplot as plt
 
 
-def plot_histogram(data, number_to_keep=False):
+def plot_histogram(data, number_to_keep=False, legend=None, options=None):
     """Plot a histogram of data.
 
-    data is a dictionary of  {'000': 5, '010': 113, ...}
-    number_to_keep is the number of terms to plot and rest is made into a
-    single bar called other values
+    Args:
+        data (list or dict): This is either a list of dictionaries containing:
+            values to represent (ex {'001': 130})
+        number_to_keep (int): DEPRECATED the number of terms to plot and rest
+            is made into a single bar called other values
+        legend(list): A list of strings to use for labels of the data.
+            The number of entries must match the lenght of data (if data is a
+            list or 1 if it's a dict)
+        options (dict): Representation settings containing
+            - number_to_keep (integer): groups max values
+            - show_legend (bool): show legend of graph content
+    Raises:
+        Exception: When legend is provided and the length doesn't match the
+            input data
     """
-    if number_to_keep is not False:
-        data_temp = dict(Counter(data).most_common(number_to_keep))
-        data_temp["rest"] = sum(data.values()) - sum(data_temp.values())
-        data = data_temp
-
-    labels = sorted(data)
-    values = np.array([data[key] for key in labels], dtype=float)
-    pvalues = values / sum(values)
-    numelem = len(values)
-    ind = np.arange(numelem)  # the x locations for the groups
-    width = 0.35  # the width of the bars
+    if options is None:
+        options = {}
+
+    if isinstance(data, dict):
+        data = [data]
+        if legend and len(legend) != 1:
+            raise Exception("Length of legendL %s doesn't match number of "
```

Comment:
We try to use functionality-specific errors, subclassing `QiskitError`. I think a `VisualizationError` class would make sense here. Do you mind to create this new class of errors and pass through other `raise` statements and change them accordingly?

Reply:
Done